pkg/controller/kogitoapp/resource/build_config_s2i.go

```diff
@@ -101,6 +113,16 @@ func setBCS2IStrategy(kogitoApp *v1alpha1.KogitoApp, buildConfig *buildv1.BuildC
 		envs = framework.EnvOverride(envs, corev1.EnvVar{Name: mavenMirrorURLEnvVar, Value: kogitoApp.Spec.Build.MavenMirrorURL})
 	}
 
+	// if user has provided a file, binary build should be used instead.
+	if kogitoApp.Spec.HasStandaloneKogitoResource() {
```

Comment:
Why not a new buildconfiguration instead of modifying this one if this if clause?

Reply:
to avoid code replication, it is basically the same thing than a usual from source build.

Reply:
We could review it and reuse the same function and decorate it to address this new build.

Well, I'll change it anyway with the Kube support, so do as you think is more convenient for now.